rules/coding-style/src/Rector/String_/UseClassKeywordForClassNameResolutionRector.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\String_;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp\Concat;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\String_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\String_\UseClassKeywordForClassNameResolutionRector\UseClassKeywordForClassNameResolutionTest
+ */
+final class UseClassKeywordForClassNameResolutionRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Use `class` keyword for class name resolution in string instead of hardcoded string reference',
+            [
+                new CodeSample(
+                    <<<'PHP'
+$this->write(
+                '<?php if (App\SomeClass::createCache($netteCacheStorage, %var,  $this->global->cacheStack, %node.array?)) { ?>',
+                md5('md5')
+            );
+PHP
+                    ,
+                    <<<'PHP'
+$this->write(
+                '<?php if (' . \App\SomeClass::class . '::createCache($netteCacheStorage, %var,  $this->global->cacheStack, %node.array?)) { ?>',
+                \md5('md5')
+            );
+PHP
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [String_::class];
+    }
+
+    /**
+     * @param String_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (substr_count($node->value, '::') !== 1) {
+            return null;
+        }
+        // a possible static call reference
+        [$before, $possibleClass, $after] = Strings::split($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#');
```

Comment:
Regex according to php documentation
https://www.php.net/manual/en/language.oop5.basic.php
>  As a regular expression, it would be expressed thus: ^[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*$.

Reply:
Best for regex is to link https://regex101.com/ example above in a comment, e.g.

```php
// @see https://regex101.com/r/cV0aRj/1/
[$before, $possibleClass, $after] = Strings::split($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#');
```

Reply:
Ready